app/controllers/api/base_controller/generic.rb

```diff
@@ -65,10 +65,10 @@ def request_retire_resource(type, id, _data = nil)
         if api_user_role_allows?('miq_request_approval')
           klass = collection_class(type)
           if id
-            msg = "Retiring #{type} id #{id} immediately as a request."
+            msg = "#{User.current_user.userid} retiring #{type} id #{id} immediately as a request."
             resource = resource_search(id, type, klass)
             api_log_info(msg)
-            klass.make_retire_request(resource.id)
+            klass.make_retire_request(resource.id, User.current_user)
```

Comment:
Kinda feels like this belongs [here](https://github.com/ManageIQ/manageiq/pull/17951/files#diff-5c80c9a5ecc28374ffccc13421a61b65R13) since it will break the UI as well, right?

Reply:
That's the requester on that line. What did I miss?

Reply:
Shouldn't it have a default value?

Reply:
No, but if it means you'll merge it, I guess it's worth the pain.

Reply:
ok. so the suggestion is to change this line back, and have the actual change over in `manageiq`?

Reply:
I don't think so. I think it makes sense to have this here, because from the API side, the requester we use is the unfortunate ```User.current_user```.